tests/phpunit/CRM/Member/Form/MembershipTest.php

```diff
@@ -1339,8 +1339,18 @@ public function testLineItemAmountOnSaleTax() {
     // ensure that the line-item values got unaffected
     $lineItem = $this->callAPISuccessGetSingle('LineItem', array('entity_id' => $membership['id'], 'entity_table' => 'civicrm_membership'));
     $this->assertEquals(1, $lineItem['qty']);
+    $this->assertEquals(50.00, $lineItem['unit_price']);
+    $this->assertEquals(50.00, $lineItem['line_total']);
     $this->assertEquals(5.00, $lineItem['tax_amount']); // ensure that tax amount is not changed
 
+    // ensure that total_amount at the Contribution level matches line_total + tax_amount at the Line Item Level
+    $contribution = $this->callAPISuccessGetSingle('Contribution',
+      array('contribution_id' => 1, 'return' => array('tax_amount', 'total_amount'),
+      )
+    );
+    $this->assertEquals($contribution['total_amount'], $lineItem['line_total'] + $lineItem['tax_amount']);
+    $this->assertEquals($contribution['tax_amount'], $lineItem['tax_amount']);
+
```

Comment:
This is the data-integrity part - we must ensure things add up. Since there is only 1 Line Item in this test we don't have to worry about summation over Line Items.